schemas.js

```diff
@@ -80,7 +80,60 @@ yup.addMethod(yup.string, "schemaFileExists", function (srcDir) {
   });
 });
 
+// validateServiceName - Custom validation method to validate service name
+yup.addMethod(yup.string, "validateServiceName", function () {
+  return this.test({
+    name: "validate-service-name",
+    test: (value, testCtx) => {
+      const alphanumericRegex = "[a-z0-9_-]+";
+      const choreoSvcRefNameRegex = new RegExp(
+        `^choreo:\/\/\/${alphanumericRegex}\/${alphanumericRegex}\/${alphanumericRegex}\/${alphanumericRegex}\/v\\d+(\\.\\d+)?\/(PUBLIC|PROJECT|ORGANIZATION)$`
+      );
+      const thirdPartySvcRefNameRegex = new RegExp(
+        "^thirdparty:[a-zA-Z0-9._/-]+$"
+      );
+      const dbSvcRefNameRegex = new RegExp("^database:[a-z0-9_/-]+$");
+
+      if (value.startsWith("choreo:///")) {
+        return (
+          choreoSvcRefNameRegex.test(value) ||
+          new yup.ValidationError(
+            `${testCtx.path} must follow the format ` +
+              `choreo:///<org-handle>/<project-handle>/<component-handle>/<endpoint-identifier>/<major-version>/<network-visibility>`
+          )
+        );
+      }
+      if (value.startsWith("thirdparty:")) {
+        return (
+          thirdPartySvcRefNameRegex.test(value) ||
+          new yup.ValidationError(
+            `${testCtx.path} has an invalid service identifier, ` +
+              `only alphanumeric characters, periods (.), underscores (_), hyphens (-), and slashes (/) are allowed after thirdparty:`
+          )
+        );
+      }
+      if (value.startsWith("database:")) {
+        return (
+          dbSvcRefNameRegex.test(value) ||
+          new yup.ValidationError(
+            `${testCtx.path} has an invalid service identifier, ` +
+              `only alphanumeric characters, underscores (_), hyphens (-), and slashes (/) are allowed after database:`
+          )
+        );
+      }
+      return (
+        thirdPartySvcRefNameRegex.test(value) ||
+        dbSvcRefNameRegex.test(value) ||
+        new yup.ValidationError(
+          `${testCtx.path} has an invalid service type. It can only contain choreo, thirdparty, or database types.`
```

Comment:
Shall we capitalize the C in Choreo here?

Reply:
@Nashaath in both places this has to be simple, as this is a predefined value by us